source/MaterialXContrib/MaterialXMaya/Plugin.cpp

```diff
@@ -288,6 +300,15 @@ MStatus uninitializePlugin(MObject obj)
             SurfaceOverride::DRAW_CLASSIFICATION,
             SurfaceOverride::REGISTRANT_ID ));
 
+    MHWRender::MRenderer* const theRenderer = MHWRender::MRenderer::theRenderer();
+    MHWRender::MFragmentManager* const fragmentManager = theRenderer ? 
+                                                         theRenderer->getFragmentManager() :
+                                                         nullptr;
+    if (fragmentManager && !fragmentManager->removeFragment(Plugin::MX_LIGHT_BUILDER_FRAG.c_str())) {
```

Comment:
I think `MShaderManager::clearEffectCache()` may be necessary for this removal to take effect.

Reply:
I agree that this would be necessary if the fragment by that name has already been used.   In practice I have found it difficult to reuse fragments with the same name.  All of this is pretty heavy handed.  A full flush of the shader cache will cause a recompile of all shaders.